app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,28 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  def do_omniauth
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    provider_data = @user.provider
+    provider_kind = @user.provider.capitalize
+
+    if @user.provider == "google_oauth2"
+      provider_kind = "Google"
+    end
+
+    if @user.persisted?
+      flash[:notice] = I18n.t("devise.omniauth_callbacks.success", kind: "#{provider_kind}")
+      sign_in_and_redirect @user, :event => :authentication
+    else
+      session["devise.#{provider_data}_data"] = request.env["omniauth.auth"]
```

Comment:
Is this data actually being used during the standard Devise registration process in some way? If we're not using it and Devise isn't using, we probably shouldn't store it.